pkg/operator/status.go

```diff
@@ -308,6 +308,18 @@ func (optr *Operator) syncUpgradeableStatus() error {
 			coStatus.Message = "One or more machine config pools are degraded, please see `oc get mcp` for further details and resolve before upgrading"
 			break
 		}
+
+		// paused required pool will prevent us from upgrading
+		if pool.Spec.Paused {
+			if _, hasRequiredPoolLabel := pool.Labels[requiredForUpgradeMachineConfigPoolLabelKey]; hasRequiredPoolLabel {
+				coStatus.Status = configv1.ConditionFalse
```

Comment:
Note that `Upgradeable=False` [only blocks minor updates][1] (4.y -> 4.(y+1)).  But :shrug:, if this is something we support, we can just document that folks should accept both "there will be no gates preventing you from triggering patch updates (4.y.z -> 4.y.z')" and "those patch updates will wedge waiting for the paused but required pool to complete".

[1]: https://github.com/openshift/api/blob/6c4f186339a7a4f505e133e9b86acd0d5aa84686/config/v1/types_cluster_operator.go#L184-L185